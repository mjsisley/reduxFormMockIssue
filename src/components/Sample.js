import React from "react";
import { reduxForm, formValueSelector, Field } from "redux-form";
import { connect } from "react-redux";

const Demo = ({ name, label }) => (
  <div
    style={{
      display: "flex",
      flexDirection: "column",
      alignItems: "center",
      width: "100%"
    }}
  >
    <label>{label}</label>
    <Field name={name} component="input" type="text" />
  </div>
);

const selector = formValueSelector("sampleform");

export class SimpleSample extends React.Component {
  render() {
    const { param, state } = this.props;
    const label = selector(state, param || "keyOfState");
    return (
      <form>
        <h1>Example Redux Form with Selector</h1>
        <Demo name="keyOfState" label={label} />
      </form>
    );
  }
}

export default connect(state => ({ state }))(
  reduxForm({
    form: "sampleform"
  })(SimpleSample)
);
