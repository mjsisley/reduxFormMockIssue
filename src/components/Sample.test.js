import React from "react";
import { Provider } from "react-redux";
import { createStore } from "redux";
import { reduxForm } from "redux-form";
import { SimpleSample } from "./Sample";
import renderer from "react-test-renderer";

function setup() {
  const spy = jest.fn();
  const store = createStore(() => ({}));
  const Decorated = reduxForm({ form: "sampleform" })(SimpleSample);

  const props = {
    param: "keyOfState",
    // state: { keyOfState: "Label" }
    state: {
      form: {
        sampleform: {
          values: { keyOfState: "Label" }
        }
      }
    }
  };
  const mockedComponent = (
    <Provider store={store}>
      <Decorated {...props}>
        <span />
      </Decorated>
    </Provider>
  );
  return mockedComponent;
}
describe("Sample Component", () => {
  it("should render the snapshot", () => {
    const mockedComponent = setup();
    const tree = renderer.create(mockedComponent).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
