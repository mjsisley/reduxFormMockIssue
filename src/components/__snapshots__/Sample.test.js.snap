// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sample Component should render the snapshot 1`] = `
<form>
  <h1>
    Example Redux Form with Selector
  </h1>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "width": "100%",
      }
    }
  >
    <label>
      Label
    </label>
    <input
      name="keyOfState"
      onBlur={[Function]}
      onChange={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      type="text"
      value=""
    />
  </div>
</form>
`;
